packages/react-focus/src/components/FocusZone/FocusZone.tsx

```diff
@@ -243,6 +243,18 @@ export class FocusZone extends React.Component<IFocusZoneProps> implements IFocu
     const { current: root } = this._root;
     const doc = this._getDocument();
 
+    // If either _activeElement or _defaultFocusElement are no longer contained by _root,
+    // reset those variables (and update tab indexes) to avoid memory leaks
+    if (
+      (this._activeElement && !elementContains(this._root.current, this._activeElement, ALLOW_VIRTUAL_ELEMENTS)) ||
+      (this._defaultFocusElement &&
+        !elementContains(this._root.current, this._defaultFocusElement, ALLOW_VIRTUAL_ELEMENTS))
+    ) {
+      this._activeElement = null;
+      this._defaultFocusElement = null;
+      this._updateTabIndexes();
+    }
```

Comment:
Should we reset both whenever either of them is removed or should there be a separate if statement for each of them (with maybe a boolean flag to indicate if one of them is reset so tab indexes are updated only once)?

Reply:
They could be updated if either of them are removed (in `updateTabIndexes`). This helps ensure that either of them aren't going to leak, in the case where one was null, the other points to a detached element, but `updateTabIndexes` doesn't update the variable holding on to the detached element.

It's more of a "safe extra-precaution"